src/jti_background.js

```diff
@@ -61,6 +64,35 @@ function saveTemplates (templateJSON, callback, responseData = null) {
     });
 }
 
+function getDomains (callback) {
+    var domainArray = {};
+    // Get the default domains
+    var domainArray2 = $.map(DefaultDomainList, function (domain, index) {
+        domain.default = true;
+        return domain;
+    });
+    // sort the default domains
+    domainArray2 = utils.sortArrayByProperty(domainArray2, 'name');
+    // Get the custom domains
+    chrome.storage.sync.get(StorageID, function (data) {
+        if (data[StorageID]) {
+            var domains = data[StorageID].options.domains;
+            domainArray = $.map(domains, function (domain, index) {
+                domain.default = false;
+                return domain;
+            });
+            // Sort them
+            domainArray = utils.sortArrayByProperty(domainArray, 'name');
+            // combine so that the default entries are always at top
+            domainArray = domainArray2.concat(domainArray);
+            callback(true, null, domainArray);
+        } else {
```

Comment:
I added the else after I had a problem where on startup the getDomains function would fail because there was no data in chrome.storage.

I did attempt a unified approach, but ran into trouble, because appending the domains if there's no data[StorageID] was adding an undefined entry partway through the list.

I could just change `domainList = domainListDefault.concat(domainList);` to assign to `domainListDefault` and use a single callback outside the `if{}else{}`, if that's all you had in mind.